### ISSUE TITLE
Travel agency tutorial: Data Index Service fail

### ISSUE THREAD

**Issue by rbaumgar** [apache/incubator-kie-examples#454]
try to run example at https://docs.jboss.org/kogito/release/latest/html_single/#con-kogito-travel-agency_kogito-deploying-on-openshift
at step 3.4.6 Installing the Kogito Data Index Service for Kogito services on OpenShift
$ kogito install data-index --infra kogito-infinispan-infra --infra kogito-kafka-infra
```
$ oc logs data-index-5b9fd7f4fb-5zgcj|more
INFO Data index persistence is set to INFINISPAN
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2020-11-13 09:27:45,155 WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.kafka.bootstrap-servers" was provided; it will be ignored; verify that the dependency extension for this configuration
 is set or you did not make a typo
ANTLR Tool version 4.7.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.7.2 used for parser compilation does not match the current runtime version 4.8ANTLR Tool ve
rsion 4.7.2 used for code generation does not match the current runtime version 4.8ANTLR Runtime version 4.7.2 used for parser compilation does not match the current runtime version 4.82020-11-13 09:27:46,820 IN
FO  [org.inf.HOTROD] (main) ISPN004021: Infinispan version: Infinispan 'Corona Extra' 11.0.3.Final
2020-11-13 09:27:47,030 ERROR [org.inf.HOTROD] (HotRod-client-async-pool-1-1) ISPN004007: Exception encountered. Retry 10 out of 10: org.infinispan.client.hotrod.exceptions.TransportException:: org.infinispan.cl
ient.hotrod.exceptions.TransportException:: ISPN004071: Connection to kogito-infinispan/172.30.43.186:11222 was closed while waiting for response.
	at org.infinispan.client.hotrod.impl.transport.netty.ActivationHandler.exceptionCaught(ActivationHandler.java:53)
	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302)
	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:281)
```
Infinispan
```
$ oc logs kogito-infinispan-0|more
2020-11-13 09:15:18,739 INFO  [io.quarkus] (main) config-generator 2.0.2.Final native (powered by Quarkus 1.5.0.Final) started in 0.007s. 
2020-11-13 09:15:18,739 INFO  [io.quarkus] (main) Profile prod activated. 
2020-11-13 09:15:18,739 INFO  [io.quarkus] (main) Installed features: [cdi, qute]
2020-11-13 09:15:19,035 INFO  [io.quarkus] (main) config-generator stopped in 0.000s
09:15:23,238 INFO  (main) [BOOT] JVM OpenJDK 64-Bit Server VM Red Hat, Inc. 11.0.9+11-LTS
09:15:23,325 INFO  (main) [BOOT] JVM arguments = [-server, -Xmx200M, -Xms200M, -XX:MaxRAM=420M, -Dsun.zip.disableMemoryMapping=true, -XX:+UseSerialGC, -XX:MinHeapFreeRatio=5, -XX:MaxHeapFreeRatio=10, -XX:+ExitOn
OutOfMemoryError, -XX:MetaspaceSize=32m, -XX:MaxMetaspaceSize=96m, -Djava.net.preferIPv4Stack=true, -Djava.awt.headless=true, -Dvisualvm.display.name=infinispan-server, -Djava.util.logging.manager=org.apache.log
ging.log4j.jul.LogManager, -Dinfinispan.server.home.path=/opt/infinispan, -classpath, :/opt/infinispan/boot/infinispan-server-runtime-11.0.4.Final-loader.jar, org.infinispan.server.loader.Loader, org.infinispan.
server.Bootstrap]
09:15:23,326 INFO  (main) [BOOT] PID = 88
09:15:23,434 INFO  (main) [org.infinispan.SERVER] ISPN080000: Infinispan Server starting
09:15:23,434 INFO  (main) [org.infinispan.SERVER] ISPN080017: Server configuration: /opt/infinispan/server/conf/infinispan.xml
09:15:23,434 INFO  (main) [org.infinispan.SERVER] ISPN080032: Logging configuration: /opt/infinispan/server/conf/log4j2.xml
09:15:27,546 INFO  (main) [org.infinispan.SERVER] ISPN080027: Loaded extension 'query-dsl-filter-converter-factory'
09:15:27,546 INFO  (main) [org.infinispan.SERVER] ISPN080027: Loaded extension 'continuous-query-filter-converter-factory'
09:15:27,628 INFO  (main) [org.infinispan.SERVER] ISPN080027: Loaded extension 'iteration-filter-converter-factory'
09:15:27,629 INFO  (main) [org.infinispan.SERVER] ISPN080027: Loaded extension 'jdk.nashorn.api.scripting.NashornScriptEngineFactory'
09:15:30,347 WARN  (main) [org.infinispan.PERSISTENCE] ISPN000554: jboss-marshalling is deprecated and planned for removal
09:15:32,444 INFO  (main) [org.infinispan.CONTAINER] ISPN000128: Infinispan version: Infinispan 'Corona Extra' 11.0.4.Final
09:15:33,328 INFO  (main) [org.infinispan.CLUSTER] ISPN000078: Starting JGroups channel infinispan with stack image-tcp
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.infinispan.commons.util.Util (file:/opt/infinispan/lib/infinispan-commons-11.0.4.Final.jar) to constructor com.sun.jndi.dns.DnsContextFactory()
WARNING: Please consider reporting this to the maintainers of org.infinispan.commons.util.Util
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
09:15:39,052 INFO  (main) [org.infinispan.CLUSTER] ISPN000094: Received new cluster view for channel infinispan: [kogito-infinispan-0-51332|0] (1) [kogito-infinispan-0-51332]
09:15:39,127 INFO  (main) [org.infinispan.CLUSTER] ISPN000079: Channel infinispan local address is kogito-infinispan-0-51332, physical addresses are [10.129.2.39:7800]
09:15:39,341 INFO  (main) [org.infinispan.CONTAINER] ISPN000390: Persisted state, version=11.0.4.Final timestamp=2020-11-13T09:15:39.249577Z
09:15:40,848 INFO  (main) [org.jboss.threads] JBoss Threads version 2.3.3.Final
09:15:41,137 INFO  (main) [org.infinispan.CONTAINER] ISPN000104: Using EmbeddedTransactionManager
09:15:43,436 INFO  (ForkJoinPool.commonPool-worker-3) [org.infinispan.SERVER] ISPN080018: Protocol HotRod (internal)
09:15:44,033 INFO  (main) [org.infinispan.SERVER] ISPN080018: Protocol REST (internal)
09:15:44,736 INFO  (main) [org.infinispan.SERVER] ISPN080004: Protocol SINGLE_PORT listening on 10.129.2.39:11222
09:15:44,736 INFO  (main) [org.infinispan.SERVER] ISPN080034: Server 'kogito-infinispan-0-51332' listening on https://10.129.2.39:11222
09:15:44,737 INFO  (main) [org.infinispan.SERVER] ISPN080001: Infinispan Server 11.0.4.Final started in 21301ms
09:15:45,550 INFO  (SINGLE_PORT-ServerIO-4-1) [org.wildfly.openssl.SSL] WFOPENSSL0002 OpenSSL Version OpenSSL 1.1.1c FIPS  28 May 2019
09:15:45,623 INFO  (SINGLE_PORT-ServerIO-4-1) [org.infinispan.SECURITY] ISPN000946: Using OpenSSL Provider
09:15:56,425 WARN  (blocking-thread--p3-t1) [org.infinispan.CONFIG] ISPN000586: The memory configuration element 'off-heap' has been deprecated. Please update your configuration
09:15:56,526 WARN  (blocking-thread--p3-t2) [org.infinispan.encoding.impl.StorageConfigurationManager] ISPN000599: Configuration for cache 'default' does not define the encoding for keys or values. If you use op
erations that require data conversion or queries, you should configure the cache with a specific MediaType for keys or values.
09:27:46,809 WARN  (SINGLE_PORT-ServerIO-4-1) [io.netty.handler.ssl.ApplicationProtocolNegotiationHandler] [id: 0x3e819742, L:/10.129.2.39:11222 ! R:/10.130.0.13:49630] TLS handshake failed: io.netty.handler.ssl
.NotSslRecordException: not an SSL/TLS record: a0031e21000003ffffffff0f0000
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1254)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1322)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:501)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:440)
```
So it looks like Date Index Service does not use SSL...

Here is the data-index-property
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: data-index-properties
  namespace: kogito-travel-agency
data:
  application.properties: >-
    infinispan.remote.sasl-mechanism=PLAIN
    infinispan.remote.server-list=kogito-infinispan:11222
    infinispan.remote.use-auth=true
    kafka.bootstrap.servers=kogito-kafka-kafka-bootstrap.kogito-travel-agency.svc:9092
    quarkus.infinispan-client.sasl-mechanism=PLAIN
    quarkus.infinispan-client.server-list=kogito-infinispan:11222
    quarkus.infinispan-client.use-auth=true
    spring.kafka.bootstrap-servers=kogito-kafka-kafka-bootstrap.kogito-travel-agency.svc:9092
```

**Comment by rbaumgar** [comment]
looks like duplicate https://github.com/kiegroup/kogito-cloud-operator/issues/607

**Comment by rbaumgar** [comment]
No, it is not a duplicate!
Problem is that no certificate is used to connect to Infinispan/Data Grid.
Found description at https://github.com/redhat-mw-demos/quarkus-dg-tls

1. Create secrets for the TLS endpoints
```
$ oc get secrets/signing-key -n openshift-service-ca -o template='{{index .data "tls.crt"}}' | openssl base64 -d -A > /tmp/server.crt
$ keytool -importcert -keystore /tmp/server.jks -storepass password -file /tmp/server.crt -trustcacerts -noprompt
$ oc create secret generic clientcerts --from-file=clientcerts=/tmp/server.jks
```
2. Update configmap data-index-properties
```
quarkus.infinispan-client.trust-store=/mnt/clientcerts
quarkus.infinispan-client.trust-store-password=password
quarkus.infinispan-client.trust-store-type=JKS
```
3. edit deployment data-index, mount volume with newly created secret
```
...
      volumes:
        - name: my-volume
          secret:
            secretName: clientcerts
            defaultMode: 420
            optional: false
...
          volumeMounts:
             - name: my-volume
              mountPath: /mnt
```
4. update configmap travel-properties and visas-properties lie data-index-properties in step 2
5. update deployment travels and visis like data-index in step 3
6. done!

**Comment by ricardozanini** [comment]
Hi! Please take a look at https://github.com/kiegroup/kogito-cloud-operator/pull/646

The fix should be delivered in the upcoming 1.0.0 version! Thanks for letting us know!

**Comment by ricardozanini** [comment]
This example was meant to be used with Infinispan 10.x version of the operator. For the new version you should wait for 1.0.0 of the Kogito Operator or configure manually as you did. :)
